test/solver_tests/edge_collapse_test.py

```diff
@@ -137,6 +138,42 @@ def test_newick_converter(self):
 
         self.assertEqual(expected_newick_string, observed_newick_string)
 
+    def test_basic_unifurcation_collapsing(self):
+
+        T = nx.DiGraph()
+        T.add_edges_from([(0, 1), (0, 2), (2, 3), (3, 4), (3, 5)])
+
+        tree = ete3.Tree(solver_utilities.to_newick(T), format=1)
+
+        collapsed_tree = solver_utilities.collapse_unifurcations(tree)
+
+        # make sure all leaves remain
+        self.assertEqual(len(tree), len(collapsed_tree))
+        for n in tree:
+            self.assertIn(n.name, collapsed_tree.get_leaf_names())
+
+        # make sure there are no singletons left
+        for n in collapsed_tree.traverse():
+            self.assertFalse(len(n.children) == 1)
```

Comment:
Maybe we should also test that the final tree is connected, and that the nodes that should be connected are connected?

Reply:
for example making sure that node 0 is connected to node 3 in the final tree?

Reply:
Done.